todocore/src/main/java/com/knighttodo/todocore/character/service/impl/ArmorServiceImpl.java

```diff
@@ -16,31 +17,28 @@
 @Service
 public class ArmorServiceImpl implements ArmorService {
 
-    private final ArmorGateway armorGateway;
     private final BonusService bonusService;
+    private final ArmorRepository armorRepository;
+    private final ArmorMapper armorMapper;
 
     @Override
     public ArmorVO save(ArmorVO armorVO) {
         armorVO.setBonuses(fetchBonusesForArmor(armorVO));
-        return armorGateway.save(armorVO);
+        return armorMapper.toArmorVO(armorRepository.save(armorMapper.toArmor(armorVO)));
```

Comment:
Looks too complicated, it's better to do it in two lines, one is the repository call and another is return mapping to the AromorVO object.